src/AutoRest.CSharp/Mgmt/Output/ResourceContainer.cs

```diff
@@ -36,24 +37,51 @@ public ResourceContainer(OperationGroup operationGroup, BuildContext<MgmtOutputL
+        public IEnumerable<ClientMethod> RemainingMethods => Methods.Where(m => m.RestClientMethod != CreateMethod
+        && m.RestClientMethod != GetRestClientMethod(ListMethod) && !ResourceOperationsListMethods.Any(r => GetRestClientMethod(r) == m.RestClientMethod));
 
-        public RestClientMethod? PutMethod => _putMethod ??= GetPutMethod();
+        public IEnumerable<PagingMethod> RemainingPagingMethods => PagingMethods.Where(m => m.Method != CreateMethod
+        && m.Method != GetRestClientMethod(ListMethod) && m.Method != GetRestClientMethod(SubscriptionExtensionsListMethod) && m.Method != GetRestClientMethod(LocationsListMethod) && !ResourceOperationsListMethods.Any(r => GetRestClientMethod(r) == m.Method));
 
-        public PagingMethod? ListMethod => _listMethod ??= FindListPagingMethod();
+        public RestClientMethod? CreateMethod => _createMethod ??= GetCreateMethod();
+
+        public ResourceListMethod? ListMethod => _listMethod ??= FindListMethod();
 
         public override ClientMethod? GetMethod => _getMethod ??= _context.Library.GetResourceOperation(OperationGroup).GetMethod;
 
-        private PagingMethod? FindListPagingMethod()
+        private ResourceListMethod? FindListMethod()
+        {
+            var listMethods = GetListMethods(true, true);
+            foreach (var listMethod in listMethods)
+            {
+                var restClientMethod = GetRestClientMethod(listMethod);
+                var pathSegments = restClientMethod?.Request.PathSegments;
+
+                // resourceGroups scope
+                if (pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/resourceGroups/"))
+                {
+
+                    return listMethod;
+                }
+            }
+
+            return null;
+        }
+
+        private RestClientMethod? GetCreateMethod()
+        {
+            return RestClient.Methods.FirstOrDefault(m => IsCreateResourceMethod(m));
+        }
+
+        private bool IsPutMethod(RestClientMethod method)
         {
-            return PagingMethods.FirstOrDefault(m => m.Method.Name.Equals("ListByResourceGroup", StringComparison.InvariantCultureIgnoreCase))
-                ?? PagingMethods.FirstOrDefault(m => m.Method.Name.Equals("List", StringComparison.InvariantCultureIgnoreCase))
-                ?? PagingMethods.FirstOrDefault(m => m.Method.Name.StartsWith("List", StringComparison.InvariantCultureIgnoreCase));
+            return method.Request.HttpMethod.Equals(RequestMethod.Put);
         }
 
-        private RestClientMethod? GetPutMethod()
+        private bool IsCreateResourceMethod(RestClientMethod method)
         {
-            return RestClient.Methods.FirstOrDefault(m => m.Request.HttpMethod.Equals(RequestMethod.Put));
+            return method.Request.HttpMethod.Equals(RequestMethod.Put) &&
+                (method.Name.Equals("CreateOrUpdate") || method.Name.Equals("Create") || method.Name.Equals("Put"));
```

Comment:
Should we also add a criteria that this method is returning a resource data? Or to be more specific, the resource data that corresponds to the current resource container

Reply:
I believe Create method doesn't always return resource data. Check few examples here - [DedicatedHosts](https://github.com/Azure/autorest.csharp/blob/feature/v3/samples/Azure.ResourceManager.Sample/Generated/DedicatedHostsRestOperations.cs#L84), [Images](https://github.com/Azure/autorest.csharp/blob/feature/v3/samples/Azure.ResourceManager.Sample/Generated/ImagesRestOperations.cs#L81), [VirtualMachines](https://github.com/Azure/autorest.csharp/blob/feature/v3/samples/Azure.ResourceManager.Sample/Generated/VirtualMachinesRestOperations.cs#L243)...etc

Above checks work fine for all the resources.